shared/src/main/scala/StringHelpers.scala

```diff
@@ -54,4 +124,54 @@ object StringHelpers:
       if index == -1 then c
       else mapping((index + 1) % mapping.length)
     }.mkString
+
+  // https://github.com/DennisMitchell/jellylanguage/blob/70c9fd93ab009c05dc396f8cc091f72b212fb188/jelly/interpreter.py#L1055
+  def sss(compressed: String)(using ctx: Context): String =
+    val decompressed = StringBuilder()
+    var integer =
+      NumberHelpers
+        .fromBase(
+          VList.from(
+            compressed.indices.map(x => CODEPAGE.indexOf(compressed(x)))
+          ),
+          250
+        ) match
+        case a: VNum => a.toInt
+        case _       => throw new Exception("InternalError: SSS failed")
+
+    while integer > 0 do
+      val mode = integer % 3
+      integer = integer / 3
+
+      if mode == 0 then
+        val code = integer % 96
+        integer = integer / 96
+        decompressed.append(CODEPAGE(code + 32))
+      else
+        var flagSwap = false
+        var flagSpace = decompressed.isEmpty
+        if mode == 2 then
+          val flag = integer % 3
+          integer = integer / 3
+          flagSwap = flag != 1
+          flagSpace = flagSpace ^ (flag != 0)
+        val useShort = integer % 2 == 0
+        integer = integer / 2
+        val words =
+          if useShort then ctx.globals.shortDictionary
+          else ctx.globals.longDictionary
+        val index = integer % words.length
+        integer = integer / words.length
+        var word = words(index)
+        if flagSwap then
+          word = (
+            word.map(c => if c.isUpper then c.toLower else c.toUpper)
+          )
```

Comment:
```suggestion
          word =
            word.map(c => if c.isUpper then c.toLower else c.toUpper)
```
Do we have a helper for toggling case? We ought to make one

Reply:
Yeah we should make one